test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java

```diff
@@ -41,10 +62,34 @@ protected Settings nodeSettings() {
 
     protected abstract void createRepository(String repoName);
 
-
-    public void testCreateSnapshot() {
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
         createRepository("test-repo");
+        final BlobStoreRepository repo = getRepository();
+        final PlainActionFuture<Void> future = PlainActionFuture.newFuture();
+        repo.threadPool().generic().execute(new ActionRunnable<>(future) {
+            @Override
+            protected void doRun() throws Exception {
+                deleteContents(repo.blobStore().blobContainer(repo.basePath()));
```

Comment:
there is no strong guarantee of clean-up on S3 with this. Do subsequent tests rely on this?

Reply:
I'm ensuring that we get an empty repo a few lines down in https://github.com/elastic/elasticsearch/pull/42653/files/574896df379b674b484fc59a75a4c55a33e7ce0b#diff-b71311294a29193871313212fcda17d5R79

Unless there is concurrent writes to the repo this should be safe. In the end we have to make the assumption that the repo is at least in a consistent (in the S3 sense of listing being consistent) state initially imo.